db/internal_stats.cc

```diff
@@ -713,6 +735,73 @@ bool InternalStats::HandleLiveSstFilesSizeAtTemperature(std::string* value,
   return true;
 }
 
+bool InternalStats::HandleNumBlobFiles(uint64_t* value, DBImpl* /*db*/,
+                                       Version* /*version*/) {
+  const auto* vstorage = cfd_->current()->storage_info();
+  const auto& blobFiles = vstorage->GetBlobFiles();
+  *value = blobFiles.size();
+  return true;
+}
+
+bool InternalStats::HandleBlobStats(std::string* value, Slice /*suffix*/) {
+  char buf[1000];
+  uint64_t numBlobFiles = 0;
+  uint64_t blobFileSize = 0;
+  HandleNumBlobFiles(&numBlobFiles, nullptr, nullptr);
+  HandleLiveBlobFileSize(&blobFileSize, nullptr, nullptr);
+  auto* currentVersion = cfd_->current();
+  uint64_t garbageSize = 0;
+  const auto& blobFiles = currentVersion->storage_info()->GetBlobFiles();
+  for (const auto& pair : blobFiles) {
+    const auto& meta = pair.second;
+    garbageSize += meta->GetGarbageBlobBytes();
+  }
+  snprintf(buf, sizeof(buf),
+           "Current version number: %s\n"
+           "Number of blob files: %s\n"
+           "Total size of blob files: %s\n"
+           "Total size of garbage in blob files: %s\n",
+           NumberToHumanString(currentVersion->GetVersionNumber()).c_str(),
+           NumberToHumanString(numBlobFiles).c_str(),
+           NumberToHumanString(blobFileSize).c_str(),
+           NumberToHumanString(garbageSize).c_str());
+  value->append(buf);
+  return true;
+}
+
+bool InternalStats::HandleTotalBlobFileSize(uint64_t* value, DBImpl* /*db*/,
+                                            Version* /*version*/) {
+  std::unordered_set<uint64_t> uniqueBlobFiles;
+  uint64_t totalBlobFileSize = 0;
+  auto* currentVersion = cfd_->current();
+  for (auto* versionCounter = currentVersion;;
+       versionCounter = versionCounter->Next()) {
+    // iterate all the versions
+    auto* vstorage = versionCounter->storage_info();
+    const auto& blobFiles = vstorage->GetBlobFiles();
+    for (const auto& pair : blobFiles) {
+      if (uniqueBlobFiles.find(pair.first) == uniqueBlobFiles.end()) {
+        // find Blob file that has not been counted
+        uniqueBlobFiles.insert(pair.first);
+        const auto& meta = pair.second;
+        totalBlobFileSize += meta->GetTotalBlobBytes();
+      }
+    }
+    if (versionCounter->Next() == currentVersion) {
+      break;
+    }
+  }
+  *value = totalBlobFileSize;
```

Comment:
I feel the actual computation belongs in `VersionSet` (see also `VersionSet::GetTotalSstFilesSize`).